src/modules/robot/arm_solutions/RotatableDeltaSolution.cpp

```diff
@@ -0,0 +1,253 @@
+#include "RotatableDeltaSolution.h"
+#include "checksumm.h"
+#include "ConfigValue.h"
+#include "ConfigCache.h"
+#include "libs/Kernel.h"
+#include "libs/nuts_bolts.h"
+#include "libs/Config.h"
+#include "libs/utils.h"
+#include "StreamOutputPool.h"
+#include <fastmath.h>
+
+#define delta_e_checksum       			CHECKSUM("delta_e_checksum")
+#define delta_f_checksum   	    		CHECKSUM("delta_f_checksum")
+#define delta_re_checksum       		CHECKSUM("delta_re_checksum")
+#define delta_rf_checksum       		CHECKSUM("delta_rf_checksum")
+#define delta_z_offset_checksum 		CHECKSUM("delta_z_offset_checksum")
+
+#define delta_ee_offs_checksum 		  	CHECKSUM("delta_ee_offs_checksum")
+#define tool_offset_checksum			CHECKSUM("tool_offset_checksum")
+
+#define z_home_angle_checksum        		CHECKSUM("z_home_angle_checksum")
+
+#define delta_printable_radius_checksum    	CHECKSUM("delta_printable_radius_checksum")
+
+#ifndef alpha_steps_per_mm_checksum
```

Comment:
The following are used in method delta_calcAngleYZ:
delta_e
delta_f
delta_re
delta_rf

and in the init():
z_calc_offset is calculate from
delta_z_offset
tool_offset
delta_ee_offs

however, the z_home_angle is not currently used, will be used in the following formula:

	z_home_offs    = (((delta_z_offset - tool_offset - delta_ee_offs) - 182.002) - 0.5);

as I have hardwired the 182.002 in the mean time to get it working, so while z_home_angle is not currently used, I’d like to keep in in there until we sort that one piece out (please!); if we don’t actually need it and that value is true for all builds, then I will remove it.  I am waiting on a reply from Neil Jensen who is rather busy trying to get the beta programme sorted.

The:
> delta_printable_radius
is not currently used but is planned for a release to enable more sophisticated checking that hot-end tool head stays within the printable area, as the PnP toolhead can move outside of it.

My intention with the ToolManager rewrite is to parameterise the allowable travel limits for tool heads.

If you want I can remove it for now and put it back later.

Cheers
Douglas

 
> On 11/06/2015, at 11:22 am, Jim Morris <notifications@github.com> wrote:
> 
> In src/modules/robot/arm_solutions/RotatableDeltaSolution.cpp <https://github.com/Smoothieware/Smoothieware/pull/673#discussion_r32177039>:
> 
> > +#include <fastmath.h>
> > +
> > +#define delta_e_checksum       			CHECKSUM("delta_e_checksum")
> > +#define delta_f_checksum   	    		CHECKSUM("delta_f_checksum")
> > +#define delta_re_checksum       		CHECKSUM("delta_re_checksum")
> > +#define delta_rf_checksum       		CHECKSUM("delta_rf_checksum")
> > +#define delta_z_offset_checksum 		CHECKSUM("delta_z_offset_checksum")
> > +
> > +#define delta_ee_offs_checksum 		  	CHECKSUM("delta_ee_offs_checksum")
> > +#define tool_offset_checksum			CHECKSUM("tool_offset_checksum")
> > +
> > +#define z_home_angle_checksum        		CHECKSUM("z_home_angle_checksum")
> > +
> > +#define delta_printable_radius_checksum    	CHECKSUM("delta_printable_radius_checksum")
> > +
> > +#ifndef alpha_steps_per_mm_checksum
> Are these needed? I don;t see them used so should be removed.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/Smoothieware/Smoothieware/pull/673/files#r32177039>.
>